text/0031-designated-operators.md

```diff
@@ -18,13 +18,46 @@ It is expected that many validators will want to rely on third parties to provis
 
 [guide-level-explanation]: #guide-level-explanation
 
-Add an additional field, expecting a public key (or some type wrapping a public key), to the `add_bid` entry point of the auction contract. The supplied public key is expected to be part of a key pair belonging to a third party ("third party" can be the validator itself, if the field is not optional). The corresponding private key will be used to sign consensus messages on the validator's behalf, as well as sign for the validator in any other instances required by the protocol proper (i.e., besides interactions with the auction contract). The link between the validator and the operator, as well as the history of validator's operator public keys, is to be tracked by the auction contract. Tracking of the pairings between a validator and its operators is intended to enable slashing, distribution of rewards and collection of transaction fees.
+At bid creation, the prospective validator may choose to specify an operator, by passing in the operator's public key. If the bid wins the auction, the operator will be responsible for creating all messages, signing with own private key. However, the validator will collect all of the rewards and would be subject to slashing, as usual. Our assumptions is that an off-chain contractual relationship provides for the operator's compensation. Existing validators and bidders are able to switch operators at will, with the change taking effect in the first future era where the new operator is specified in a validator's winning bid. There is no confirmation for the relationship, but provisions have to be made to restrict designation to simplify reasoning about the behavior of the validator-operator pair. 
```

Comment:
"assumptions is" → "assumption is"

> switch operators at will, with the change taking effect in the first future era where the new operator is specified in a validator's winning bid

I just wanted to say that:

> ~Actually, this is probably something that could even take effect sooner, right after the next era change? It wouldn't change the validator weights, after all.~

But currently the leader schedule also depends on the _order_ of validator keys (lexicographically), which _would_ change if consensus only knows about the operator keys. :confused:

Reply:
But we could make the consensus "key" just the tuple of both keys, with the validator key first and the operator key second. For cryptography it would only use the operator key, but the ordering would be determined by the validator key.

On the other hand, it's probably surprising behavior if `add_bid` takes effect partly in the next era and partly `auction_delay` eras in the future.

Reply:
Currently consensus doesn't know anything about bid state until the era where new winners take over begins, right? I am not suggesting any changes to that. If you want a new operator, you have to wait a while for it to make a difference in consensus, just as with changing the amount of tokens you bid.